### DISCUSSION TITLE
Upgrading from 3.14.3 to 3.20.4 or 4.14.2 breaks global types

### DISCUSSION THREAD

**Discussion by akrapacs** [panva/jose#533] [Q&A] [answered]
Project is using node 18.12.1 and on `jose` 3.14.3. When I update to 3.20.4 it seems to break type resolution for global types used in the project and dependencies. I think it is just with `URL` and `URLSearchParams` but I'm not sure because there's quite a lot of errors in the build output:

`error TS2304: Cannot find name 'URLSearchParams'`
`error TS2552: Cannot find name 'URL'`

I tried updating to 4.14.2 with the same result but an additional problem with the following that I need seeming not available in version 4.x

`import { parseJwk } from 'jose/jwk/parse';`

I don't see a `parseJwk` equivalent after changing the import to `import * as jose from 'jose'`;

I would like to update to the latest version but in both cases I don't know how to resolve the either of these problems.

**Comment by panva** [comment, marked as answer]
> I don't see a `parseJwk` equivalent after changing the import to `import * as jose from 'jose'`;

Have you looked at the [v4.0.0 release notes](https://github.com/panva/jose/releases/tag/v4.0.0), especially the migration guide and breaking changes?

As far as global types, you're not running correct/latest @types/node?

**Reply by akrapacs** [reply]
Yes thank you I was looking for for a migration guide or breaking changes but I wasn't able to find it, that solved the issue with `parseJwk`. And installing the latest @types/node did solve the other issue as well. I appreciate the swift response!
